src/niamoto/core/plugins/widgets/donut_chart.py

```
@@ -0,0 +1,396 @@
+
+    # get_container_html is inherited from WidgetPlugin
+
+    def render(self, data: Optional[Any], params: DonutChartParams) -> str:
+        """Generate the HTML for the donut chart or subplots."""
+        # Debug output
+        print("\n" + "=" * 80)
+        print("DEBUG DONUT_CHART - params:", params)
+        print("DEBUG DONUT_CHART - data type:", type(data))
+        if isinstance(data, dict):
+            print("DEBUG DONUT_CHART - data keys:", list(data.keys()))
+            for key in data.keys():
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)